mass-deployer/internal/parser/parser.go

```diff
@@ -0,0 +1,103 @@
+package parser
+
+import (
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/cosmos/go-bip39"
+	deployer "github.com/threefoldtech/tfgrid-sdk-go/mass-deployer/pkg/mass-deployer"
+	"gopkg.in/yaml.v3"
+)
+
+func ParseConfig(configFile []byte) (deployer.Config, error) {
+	conf := deployer.Config{}
+	nodeGroupsNames := []string{}
+
+	err := yaml.Unmarshal(configFile, &conf)
+	if err != nil {
+		return deployer.Config{}, err
+	}
+
+	mnemonic := conf.Mnemonic
+	network := conf.Network
+
+	if strings.TrimSpace(mnemonic) == "" {
```

Comment:
https://pkg.go.dev/gopkg.in/validator.v2 nonzero validator can help alot to remove some of the code repeatition